tests/MongoDB.Driver.Core.Tests/Core/Operations/RetryabilityHelperTests.cs

```diff
@@ -45,18 +45,24 @@ public void AddRetryableWriteErrorLabelIfRequired_should_add_RetryableWriteError
         {
             var exception = CoreExceptionHelper.CreateMongoWriteConcernException(BsonDocument.Parse($"{{ writeConcernError : {{ code : {errorCode} }} }}"));
 
-            RetryabilityHelper.AddRetryableWriteErrorLabelIfRequired(exception);
+            RetryabilityHelper.AddRetryableWriteErrorLabelIfRequired(exception, Feature.ServerReturnsRetryableWriteErrorLabel.LastNotSupportedVersion);
 
             var hasRetryableWriteErrorLabel = exception.HasErrorLabel("RetryableWriteError");
             hasRetryableWriteErrorLabel.Should().Be(shouldAddErrorLabel);
         }
 
-        [Fact]
-        public void AddRetryableWriteErrorLabelIfRequired_should_add_RetryableWriteError_for_network_errors()
+        [Theory]
+        [ParameterAttributeData]
+        public void AddRetryableWriteErrorLabelIfRequired_should_add_RetryableWriteError_for_network_errors([Values(false, true)] bool isNewBehavior)
         {
             var exception = (MongoException)CoreExceptionHelper.CreateException(typeof(MongoConnectionException));
 
-            RetryabilityHelper.AddRetryableWriteErrorLabelIfRequired(exception);
+            var serverReturnsRetryableWriteErrorLabelFeature = Feature.ServerReturnsRetryableWriteErrorLabel;
+            RetryabilityHelper.AddRetryableWriteErrorLabelIfRequired(
+                exception, 
+                isNewBehavior 
+                    ? serverReturnsRetryableWriteErrorLabelFeature.FirstSupportedVersion
+                    : serverReturnsRetryableWriteErrorLabelFeature.LastNotSupportedVersion);
```

Comment:
Move argument setup to the Arrange section keeping the Act section as simple as possible. Refactor lines 58-68 to:

    var exception = (MongoException)CoreExceptionHelper.CreateException(typeof(MongoConnectionException));
    var feature = Feature.ServerReturnsRetryableWriteErrorLabel;
    var serverVersion = serverReturnsRetryableWriteErrorLabel ? feature.FirstSupportedVersion : feature.LastNotSupportedVersion;

    RetryabilityHelper.AddRetryableWriteErrorLabelIfRequired(exception, serverVersion);

    exception.HasErrorLabel("RetryableWriteError").Should().BeTrue();